collect_app/src/main/java/org/odk/collect/android/formentry/FormEntryViewModel.java

```diff
@@ -168,6 +176,13 @@ private String getFormIdentifierHash() {
         }
     }
 
+    private void ensureFormController() {
+        if (this.formController == null) {
+            Timber.w("Null formController");
```

Comment:
I'm wondering if we want to add any more logging here. It'd be great to try and get as much detail as possible when this happens.